api/src/Controller/Event/DeleteEvent.php

```diff
@@ -47,9 +47,11 @@ public function buildResource(Request $request, Response $response, $params): ar
         $permissions = ['api.delete.event'];
         $this->checkPermissions($permissions);
         $this->getEvent($params['id']);
-
-        $delete = Delete::new($this->container->db);
-        $result = $delete->from('Events')->whereEquals(['EventID' => $params['id']])->perform();
+        try {
+            $this->container->EventService->deleteById($params['id']);
+        } catch (\Exception $e) {
+            throw new InvalidParameterException($e->getMessage());
+        }
```

Comment:
I think we can probably avoid the use of try/catch here if we have the `EventRepository` throwing `InvalidParameterException` instead of `Exception`, right? I think we set up the error handler middleware to handle those cases with the correct HTTP Status.